### ISSUE TITLE
Modifying MANIAC-V2 corrupts it

### ISSUE THREAD

**Issue by dwatteau** [dwatteau/scummtr#16]
When just applying this to MANIAC-PC-V2, without any other change:

```sh
scummrp -gp maniacv2 . -od DUMP # export
scummrp -gp maniacv2 . -id DUMP # reimport the same dump
```

…`scummrp` seems to corrupt the game, because right after this, the game always starts in demo mode, at least in ScummVM. This problem also happens with the 0.4.0 `scummrp.exe` executable.

With a more verbose export, the following warnings appear:

```
17.LFL should actually end at 0x1F3B
Removed (33, 0x3E77) from the index
Removed (41, 0x160F) from the index
OIv2_0395 might actually be OIv2_0547 (unlikely)
```

but the first warning about `17.LFL` is different on the French version of Maniac Mansion v2.

```
Removed (17, 0x1F3B) from the index
Removed (33, 0x3E77) from the index
Removed (41, 0x160F) from the index
OIv2_0395 might actually be OIv2_0547 (unlikely)
```

It appears that the original English v2 version of Maniac Mansion has more script bugs than the one already discovered in #12.

This ScummRP bug is potentially responsible for the fact that PR #14 still doesn't completely fix our ScummTR bug with the same variant of the game.

**Comment by dwatteau** [comment]
So, the bug is not only in the English V2 version of Maniac Mansion; I can also reproduce it with my French V2 version (included in my DOTT CDs). I think some Italian translators hit this bug, too.

**Comment by dwatteau** [comment]
There's probably something wrong in the V2 parser. Zak appears OK, but it still gives some warnings and has some `FIXME`s. Reimporting an official translation inside Zak also produces different LFL files.

AFAIK, scummtr always behaved this way.

So, we probably have a bug or some limitations here, but I don't know how to fix that. ScummTR 0.5.1 will just disable support for MANIAC-V2 modifications.

**Comment by dwatteau** [comment]
Someone on oldgamesitalia reported that this bug makes the Italian Amiga version glitch in some way, unless you fix some offsets:

https://www.oldgamesitalia.net/forum/index.php?s=&showtopic=18982&view=findpost&p=385373

> to fix the bug [with the Italian Maniac V2 Amiga version], you must substitute the following 4 bytes in file 45.LFL
>
> offset `2a`: from `FF` to `92`
> offset `2b`: from `FF` to `eb`
> offset `2c`: from `FF` to `92`
> offset `2d`: from `FF` to `eb`
>
> basically, xor with ff (e.g. $eb92 xor $ffff = $146d) the offsets from which the graphics of object 8 and object 9 that make up the animation of the first character start
>
> I got the correct values from the English version (or some other version)

**Comment by dwatteau** [comment]
Note: actually the bug where the game goes into demo mode once modified… maybe just comes from ScummVM fallback detection code being wrong somewhere in V2 — I have a memory of that Catalan translation possibly hitting the bug before scumm-md5 was updated for it

**Comment by Jibun** [comment]
I don’t know if I can shed some light on this issue, but here’s what I remember from dealing with this bug while working on the Catalan version of the game:

I’m not sure if this is a bug in the tool or in the game files, but in the Catalan version I fixed an error with _scummrp_ by following the guides of a previous modder who made an improved Spanish translation of the game.

The fix was applied to `7.LFL`, specifically to `LFv2_0007/SCv2_0059` (it had a Spanish sentence, but I simply added the translation on top).
I have no fixes for `17.LFL`

I used **ScummRp 0.2** by Thomas Combeleran. I couldn’t use your latest version because it was capped for _maniacv2_. I don’t remember if there were other bugs in previous versions, which may be why I ended up using Thomas’s original release.

If I have some time this afternoon, I’ll try repackaging the game files with some of your versions of the tool to see if it produces any errors.

**Comment by Jibun** [comment]
Ok, so I’ve been doing some testing. Here are my results:

The **0.4.2** version behaves the same as **0.2** with respect to MMV2, and repackages it with the following messages:
```
17.LFL should actually end at 0x1F3B
Regarding LFv2_0033:
Removed (33, 0x3E77) from the index
Regarding LFv2_0041:
Removed (41, 0x160F) from the index
Regarding LFv2_0045:
OIv2_0395 might actually be OIv2_0547 (unlikely)
```
The **0.5.0** version returns the following messages and does not repackage:
```
Fixing unexpected size at 0x0 in 01.LFL
ERROR: Bad OIv2 offset
```
From **0.5.1** onwards, compatibility with MMV2 is disabled.

I’ve managed to successfully repackage the game with versions **0.2** and **0.4.2**. I haven’t found any major corruption. The only thing I always have to fix manually after repackaging—and that may be related to the message `OIv2_0395 might actually be OIv2_0547 (unlikely)`—is Dave’s animation in the character selection screen.
In file **45.LFL**, I always have to copy the 2 bytes starting at position 40 to positions 42 and 44, since for some reason the repackager just puts zeros there (then masked to F’s).

In conclusion, there doesn’t seem to be anything major to fix, at least in the earlier versions of this tool.

**Comment by dwatteau** [comment]
@Jibun Thank you very much for looking into this, testing various releases, and sharing your results here. It's very much appreciated.

Hopefully I'll have some time/motivation to look into some Maniac V2 matters again. Stay tuned!
